pkg/utils/kubernetes/resources/deployments.go

```diff
@@ -121,11 +129,17 @@ func GenerateNewDeploymentForControlPlane(params GenerateNewDeploymentForControl
 	return deployment, nil
 }
 
+// GenerateContainerForControlPlaneParams is a parameter struct for GenerateControlPlaneContainer function.
+type GenerateContainerForControlPlaneParams struct {
+	Image                          string
+	AdmissionWebhookCertSecretName string
```

Comment:
How about using `mo.Option` or just a pointer to signal this is optional?

Reply:
👍 I'll go with a pointer just for the sake to not warrant a dependency on `mo` since we do not use that just yet in this project